specifyweb/frontend/js_src/lib/components/QueryBuilder/Wrapped.tsx

```diff
@@ -149,8 +155,17 @@ export function QueryBuilder({
     'showHiddenFields'
   );
 
-  const isResourceModified = useIsModified(queryResource);
-  const saveRequired = isResourceModified || state.saveRequired;
+  const [saveRequired, setSaveRequired] = React.useState(false);
+  const checkForChanges = throttle(() => {
+    const initialFieldsObject = JSON.parse(initialFields.current);
+    const stateFieldsObject = JSON.parse(JSON.stringify(state.fields));
+    setSaveRequired(
+      JSON.stringify(initialFieldsObject) !== JSON.stringify(stateFieldsObject)
+    );
+  }, 200);
+
+  React.useEffect(checkForChanges, [state.fields]);
+
```

Comment:
can you move all of this code to bear near line 135?
it's because this code is related to `initialFields`, so it's best to keep related code closer to each other - easier to read the code that way